### ISSUE TITLE
For CIME cases, how do we add constant initial conditions?

### ISSUE THREAD

**Issue by AaronDonahue** [E3SM-Project/scream#2573]
This is less of an issue and more of a question.  In the scream_input.yaml file we are able to specify a constant initial value for a field by adding it to the list of args under `initial_conditions`.  We use that to initialize things like `qm`.  There is already a list of these in the `namelist_defaults.xml` file in the repo, (see example below) but given the way these are designated in the namelist file it isn't clear how you can add a new one "on the fly".  My suspicion is you can't.

```  
<initial_conditions>
    <Filename type="file">/usr/gdata/climdat/ccsm3data/inputdata/atm/scream/init/screami_ne30np4L72_20220823.nc</Filename>
    <topography_filename type="file">/usr/gdata/climdat/ccsm3data/inputdata/atm/cam/topo/USGS-gtopo30_ne30np4pg2_x6t-SGH.c20210614.nc</topography_filename>
    <restart_casename type="string">SMS_Ld1_P216x1.ne30pg2_ne30pg2.F2010-SCREAMv1.quartz_intel.test_u_v_U_V.scream</restart_casename>
    <surf_evap type="real">0.0</surf_evap>
    <precip_liq_surf_mass type="real">0.0</precip_liq_surf_mass>
    <precip_ice_surf_mass type="real">0.0</precip_ice_surf_mass>
 ...
 ```

**Comment by bartgol** [comment]
Yeah, we don't really have a support for that. I think we may be able to do so by changing the way we organize it. Namely, instead of having
```
initial_conditions:
  field_1: val_1
  field_2: val_2
```
we may do it via
```
  constant_fields: [field_1, field_2, ...]
  constant_values: [val_1, val_2, ...]
```
The drawback is that this format would not allow to init a vector field with different values for its components. E.g., I don't think you'd be able to init `my_vec_field` with `[1, 0]`, like you are able to do now. If this is a limitation, we can think of cooking up some special syntax, like
```
 constant_fields: [my_vec_field@cmp=0, my_vec_field@cmp=1]
 constant_values: [1,0]
```

I can work on something along these lines, if this is needed.

**Comment by AaronDonahue** [comment]
I don't think it is high priority.  I only hit it because I was experimenting with something.  But I could see it being something we may want to control in the future.  More from the perspective of wanting to delete one of the pre-set constant fields.
